pkg/apis/sparkoperator.k8s.io/v1beta1/types.go

```diff
@@ -385,6 +385,9 @@ type SparkPodSpec struct {
 	// Sidecars is a list of sidecar containers that run along side the main Spark container.
 	// Optional.
 	Sidecars []apiv1.Container `json:"sidecars,omitempty"`
+	// NodeSelector is the Kubernetes node selector to be added to the driver and executor pods.
+	// Optional.
+	NodeSelector map[string]string `json:"nodeSelector,omitempty"`
```

Comment:
The top-level `SparkApplicationSpec` also has a `NodeSelector` field that applies to both driver and executor pods as the Kubernetes backend doesn't support separate node selector config for driver and executors. This new field should be mutually exclusive to the top-level one. Let's make this clear in the comment both here and in place for the top-level field. And let's add a deprecation comment for the top-level field.

Reply:
The idea that I had in mind it that was additive (if any at Pod level, would be added to what was declared at app spark level). I've pushed the changes in the doc, and also a validator at the controller level. Doing a new transition to failed if the validation is not passed (was not very sure about this part, please double check).